src/metabase/upload.clj

```diff
@@ -589,10 +590,59 @@
   [db table]
   (nil? (can-update-error db table)))
 
+(defn- check-can-delete
+  "Throws an error if the given table is not an upload, or if the user does not have permission to delete it."
+  [db table]
+  ;; For now anyone that can update a table is allowed to delete it.
+  (when-let [error (can-update-error db table)]
+    (throw error)))
+
 ;;; +--------------------------------------------------
 ;;; |  public interface for updating an uploaded table
 ;;; +--------------------------------------------------
 
+(def ^:private table-referencing-models
```

Comment:
maybe move this and `delete-table-from-app-db!` up a bit, these are not public interface